Activities/Cryptography/UiPath.Cryptography.Activities/DecryptFile.cs

```diff
@@ -90,43 +105,54 @@ protected override void Execute(CodeActivityContext context)
         {
             try
             {
-                string inputFilePath = InputFilePath.Get(context);
-                string outputFilePath = OutputFilePath.Get(context);
-                string key = Key.Get(context);
-                SecureString keySecureString = KeySecureString.Get(context);
-                Encoding keyEncoding = KeyEncoding.Get(context);
-
-                if (string.IsNullOrWhiteSpace(inputFilePath))
-                {
+                var inputFilePath = InputFilePath.Get(context);
+                var inputFile = InputFile.Get(context);
+                var outputFilePath = OutputFilePath.Get(context);
+                var key = Key.Get(context);
+                var keySecureString = KeySecureString.Get(context);
+                var keyEncoding = KeyEncoding.Get(context);
+
+                if (string.IsNullOrWhiteSpace(inputFilePath) && inputFile == null)
                     throw new ArgumentNullException(Resources.InputFilePathDisplayName);
-                }
-                if (string.IsNullOrWhiteSpace(outputFilePath))
-                {
+
+                if (string.IsNullOrWhiteSpace(outputFilePath) && inputFile == null)
                     throw new ArgumentNullException(Resources.OutputFilePathDisplayName);
-                }
+
+                //either input file path or input file as resource should be used
+                if (!string.IsNullOrWhiteSpace(inputFilePath) && inputFile != null)
+                    throw new ArgumentNullException(string.Format(Resources.Exception_UseOnlyFilePathOrInputResource,
```

Comment:
`ArgumentNullException` is misleading, as it's only supposed to be used when an argument is null.
Let's throw plain `ArgumentException` instead.

Reply:
fixed